GitCommands/CommitHelper.cs

```diff
@@ -4,74 +4,83 @@ namespace GitCommands
+            using (var textWriter = new StreamWriter(file, false, module.CommitEncoding))
             {
-                textWriter.Write(commitMessageText);
+                textWriter.Write(messageText ?? string.Empty);
             }
 
             if (AppSettings.RememberAmendCommitState && amendCommit)
             {
                 File.WriteAllText(GetAmendPath(module), true.ToString());
             }
-            else if (File.Exists(GetAmendPath(module)))
+            else
             {
                 File.Delete(GetAmendPath(module));
             }
         }
 
-        public static string GetCommitMessage(GitModule module)
+        private static string GetAmendPath(GitModule module)
         {
-            if (File.Exists(GetCommitMessagePath(module)))
-            {
-                return File.ReadAllText(GetCommitMessagePath(module), module.CommitEncoding);
-            }
-
-            return string.Empty;
+            return GetFilePath(module, "GitExtensions.amend");
         }
 
-        public static string GetCommitMessagePath(GitModule module)
+        private static bool GetAmendState(GitModule module)
         {
-            return GetFilePath(module, "COMMITMESSAGE");
+            bool amendState = false;
+
+            if (AppSettings.RememberAmendCommitState)
+            {
+                string amendSaveStateFilePath = GetAmendPath(module);
+                if (File.Exists(amendSaveStateFilePath))
+                {
+                    bool.TryParse(File.ReadAllText(amendSaveStateFilePath), out amendState);
+                }
+            }
+
+            return amendState;
         }
 
-        private static string GetAmendPath(GitModule module)
+        private static string GetFilePath(GitModule module, string fileName)
         {
-            return GetFilePath(module, "GitExtensions.amend");
+            return Path.Combine(module.WorkingDirGitDir, fileName);
         }
 
-        private static string GetFilePath(GitModule module, string action)
+        private static string GetMergeMessagePath(GitModule module)
         {
-            return Path.Combine(module.WorkingDirGitDir, action);
+            return GetFilePath(module, "MERGE_MSG");
         }
 
-        public static bool GetAmendState(GitModule module)
+        public static (string FilePath, bool FileExists) GetMergeOrCommitMessagePath(GitModule module)
         {
-            bool amendState = false;
-
-            if (AppSettings.RememberAmendCommitState && File.Exists(GetAmendPath(module)))
+            string mergeMessagePathName = GetMergeMessagePath(module);
+            if (File.Exists(mergeMessagePathName))
             {
-                var amendSaveStateFilePath = GetAmendPath(module);
-                bool.TryParse(File.ReadAllText(amendSaveStateFilePath), out amendState);
-                try
-                {
-                    File.Delete(amendSaveStateFilePath);
```

Comment:
This was unexpected, unnecessary and simply wrong in the case of a crash in FormCommit.